bigquery_etl/query_scheduling/task.py

```diff
@@ -332,6 +333,19 @@ def of_query(cls, query_file, metadata=None, dag_collection=None):
         if metadata.is_public_json():
             task_config["public_json"] = True
 
+        # Override the table_partition_template if the bq partition type is MONTH
+        # Pass a jinja template that reformats the date string used for table partition name.
+        if metadata.bigquery and metadata.bigquery.time_partitioning:
+            destination_table = metadata.scheduling.get("destination_table")
+            # only proceed if there is no destination_table specified
+            if (
+                destination_table is None
+                and metadata.bigquery.time_partitioning.type is PartitionType.MONTH
+            ):
+                task_config[
+                    "table_partition_template"
+                ] = '{{ macros.ds_format(ds, "%Y-%m-%d", "%Y%m") }}'
+
```

Comment:
We can probably add support for `DAY`, `YEARLY` and `HOURLY` here as well. Or throw an error that they aren't supported at the moment if they are used

Reply:
Thanks for the review, I went ahead and implemented that change. would appreciate a second set of eyes on this since it now touches every dag.